actions.js

```diff
@@ -0,0 +1,234 @@
+const CHOICES_SOURCES = [
+	{ label: 'PHONO', id: 'PHONO' },
+	{ label: 'CD', id: 'CD' },
+	{ label: 'TUNER', id: 'TUNER' },
+	{ label: 'DVD', id: 'DVD' },
+	{ label: 'BluRay', id: 'BD' },
+	{ label: 'HDP', id: 'HDP' },
+	{ label: 'TV', id: 'TV' },
+	{ label: 'TV/CBL', id: 'TV/CBL' },
+	{ label: 'SAT/CBL', id: 'SAT/CBL' },
+	{ label: 'SAT', id: 'SAT' },
+	{ label: 'Media Player', id: 'MPLAY' },
+	{ label: 'Game', id: 'GAME' },
+	{ label: 'HD Radio', id: 'HDRADIO' },
+	{ label: 'NET/Online', id: 'NET' },
+	{ label: 'Pandora', id: 'PANDORA' },
+	{ label: 'SiriusXM', id: 'SIRIUSXM' },
+	{ label: 'Spotify', id: 'SPOTIFY' },
+	{ label: 'LastFM', id: 'LASTFM' },
+	{ label: 'Flickr', id: 'FLICKR' },
+	{ label: 'iRadio', id: 'IRADIO' },
+	{ label: 'Server', id: 'SERVER' },
+	{ label: 'Favorites', id: 'FAVORITES' },
+	{ label: 'VCR', id: 'VCR' },
+	{ label: 'DVR', id: 'DVR' },
+	{ label: 'V.AUX', id: 'V.AUX' },
+	{ label: 'Aux1', id: 'AUX1' },
+	{ label: 'Aux2', id: 'AUX2' },
+	{ label: 'Aux3', id: 'AUX3' },
+	{ label: 'Aux4', id: 'AUX4' },
+	{ label: 'Bluetooth', id: 'BT' },
+	{ label: 'NET/USB', id: 'NET/USB' },
+	{ label: 'XM', id: 'XM' },
+	{ label: '8K', id: '8K' },
+	{ label: 'IPOD', id: 'IPOD' },
+	{ label: 'Same as zone 1', id: 'SOURCE' },
+]
+
+module.exports = function (self) {
+	CHOICES_VOLUME = []
+
+	for (i = 98; i >= 1; i--) {
+		db = i - 80
+		v = String(i).padStart(2, '0')
+		CHOICES_VOLUME.push({ label: db + 'dB', id: '' + v + '' })
+	}
+
+	CHOICES_VOLUME.push({ label: 'Mute', id: '00' })
+
+	self.setActionDefinitions({
+		power: {
+			name: 'Power control',
+			options: [
+				{
+					type: 'dropdown',
+					label: 'on/standby',
+					id: 'power',
+					default: 'PWON\r',
+					choices: [
+						{ label: 'Power on', id: 'PWON' },
+						{ label: 'standby', id: 'PWSTANDBY' },
+						{ label: 'Zone2 on', id: 'Z2ON' },
+						{ label: 'Zone2 off', id: 'Z2OFF' },
+						{ label: 'Zone3 on', id: 'Z3ON' },
+						{ label: 'Zone3 off', id: 'Z3OFF' },
+					],
+				},
+			],
+			callback: async (event) => {
```

Comment:
I used the asynch line as it was written in the module template file.  I've removed asynch from all of the actions in this module.  I'm not clear on the real-world implications here -- This may be something that's worth discussing on a wiki page? (when to use asynch or why) -- and/or change that line in the template file to keep others like me from assuming this is the way it's supposed to be done?  (I've done a LOT of assuming)